src/main/java/htsjdk/samtools/cram/build/ContainerFactory.java

```diff
@@ -77,116 +65,51 @@ Container buildContainer(final List<CramCompressionRecord> records,
         for (int i = 0; i < records.size(); i += recordsPerSlice) {
             final List<CramCompressionRecord> sliceRecords = records.subList(i,
                     Math.min(records.size(), i + recordsPerSlice));
-            final Slice slice = buildSlice(sliceRecords, header);
+            final Slice slice = Slice.buildSlice(sliceRecords, header);
             slice.globalRecordCounter = lastGlobalRecordCounter;
             lastGlobalRecordCounter += slice.nofRecords;
             container.bases += slice.bases;
             slices.add(slice);
-
-            // assuming one sequence per container max:
-            if (container.sequenceId == -1 && slice.sequenceId != -1)
-                container.sequenceId = slice.sequenceId;
         }
 
-        container.slices = slices.toArray(new Slice[slices.size()]);
+        container.slices = slices.toArray(new Slice[0]);
         calculateAlignmentBoundaries(container);
 
         globalRecordCounter += records.size();
         return container;
     }
 
-    private static void calculateAlignmentBoundaries(final Container container) {
+    static void calculateAlignmentBoundaries(final Container container) {
```

Comment:
to make it testable

Reply:
Excellent points.  Added commit 5447ca665b9f87db9355b2ccefb233e82f13b915 for this.